include/zephyr/net/wifi_mgmt.h

```diff
@@ -145,6 +145,8 @@ enum net_event_wifi_cmd {
 	NET_EVENT_WIFI_CMD_RAW_SCAN_RESULT,
 	/** Disconnect complete */
 	NET_EVENT_WIFI_CMD_DISCONNECT_COMPLETE,
+	/** Wi-Fi NM events, payload depends on NM **/
+	NET_EVENT_WIFI_CMD_NM_EVENT,
```

Comment:
I'm not really convinced that such an event belongs in the generic `wifi_mgmt` layer, IMO it's way too abiguous. I think that we should rather:
* Define own NET_EVENT space for supplicant-specific events, outside of generic header,
* or follow the pattern used in sensor API and recently in 802.15.4 radio API - define a new enum in the "generic" part of the wifi_mgmt which indicates end of the public events (see https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/sensor.h#L197 for example) and let other components use it to define extensions, i. e. new events.

Reply:
Done

Reply:
I don't think my previous comment was understood well. What I meant, is that instead of introducing vague events into generic part of the `wifi_mgmt`, we could add here (to `enum net_event_wifi_cmd`):
* `NET_EVENT_WIFI_CMD_MAX`
* `NET_EVENT_WIFI_CMD_PRIV_START = NET_EVENT_WIFI_CMD_MAX`

This leaves an opening to external modules (like supplicant) to add custom Wi-Fi level events in their own headers, w/o altering the "common" space, for example:

```c
enum net_event_wifi_supp {
	NET_EVENT_WIFI_SUPP_CONNECTED = NET_EVENT_WIFI_CMD_PRIV_START ,
	NET_EVENT_WIFI_SUPP_DISCONNECTED ,
	NET_EVENT_WIFI_SUPP_ASSOC_REJECT,
	etc.
}
```
WDYT?

Reply:
Sensors did this too, and it works quite well

Reply:
We support multiple NM's at the same time, so, how does this work then? One simplest way would be to interleave e.g., `priv_start, priv_end=priv_start+256` and then `priv2_start=priv_end`..

Reply:
Yes, with multiple NM's they would need to use custom events (different range). Or, if that's a conern, follow my initial suggestion:
> Define own NET_EVENT space for supplicant-specific events, outside of generic header

NM's could define their own event space (i. e. not using `#define _NET_WIFI_CODE	0x156` but a custom one).

In any case, having a single `NET_EVENT_WIFI_CMD_NM_EVENT` event doesn't seem to solve multi-nm issue either, as you never know which one triggered the event.